plugin/plugin.go

```diff
@@ -0,0 +1,129 @@
+// Copyright 2024 the Drone Authors. All rights reserved.
+// Use of this source code is governed by the Blue Oak Model License
+// that can be found in the LICENSE file.
+
+package plugin
+
+import (
+	"archive/zip"
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+type ZipPlugin struct {
+	Source   string `envconfig:"PLUGIN_SOURCE"`
+	Target   string `envconfig:"PLUGIN_TARGET"`
+	LogLevel string `envconfig:"PLUGIN_LOG_LEVEL"`
+}
+
+func (p *ZipPlugin) Exec(ctx context.Context) error {
```

Comment:
Also, let’s move the zip-specific code to a separate gzip folder, similar to the [cache plugin](https://github.com/drone-plugins/drone-meltwater-cache/tree/master/archive). We can follow the same approach for the new format types we’ll be adding next.

Reply:
Done